### ISSUE TITLE
Windows Node Support

### ISSUE THREAD

**Issue by jnoller** [Azure/AKS#980]
Windows Server Containers is in GA, as well as windows node support entering GA in Kubernetes 1.14. This is a tracking issue for Windows worker node support in AKS

**Comment by jnoller** [comment]
Windows worker node support is now in public preview: https://github.com/Azure/AKS/blob/master/previews.md#windows-worker-nodes

https://aka.ms/aks/windows
https://aka.ms/aks/windowsdocs

**Comment by nmehlei** [comment]
Is there an estimated GA date? Probably depending on the number and complexity of upcoming issues but at least a rough estimate would be helpful.

**Comment by jnoller** [comment]
@nmehlei we do not commit to specific dates for delivery. I can say that features in preview normally soak for at least 4 weeks or more until GA

**Comment by MikkelHegn** [comment]
Hi @nmehlei - what's your timeline and criteria for using this feature? I'm interested in knowing as this might differ as to what we look at for GA.

**Comment by jimoneil** [comment]
Is there a more detailed timeline of support or features? For instance, do we expect full parity in terms of support for kubenet at GA?

**Comment by nmehlei** [comment]
@MikkelHegn We are currently evaluating it as the basis for a few of our planned production systems, which are planned to go online in August/September this year. Obviously, we're very skeptical about doing this if it is still in (Public) Preview. 

Our evaluation currently did not yet go far enough to specify which (additional) features we will need but in general, we need an infrastructure to host multiple clustered Docker-based services in a highly-available manner with attached persistent storage.

**Comment by dariuszbz** [comment]
I'm lost. Is it GA or still preview? Can i run production system or I need wait until GA? Clear message please :) Thank You so much.

**Comment by MikkelHegn** [comment]
The issue is not closed, so the work is not done, so it's not GA :-)

**Comment by dariuszbz** [comment]
Oooooook.

When do you plan to release GA?

**Comment by kkopuru** [comment]
AKS 1.14 is in GA.

When can we expect Windows node support in GA.

**Comment by chintandalwadiwk** [comment]
@jnoller :  It is very helpful , if we have a tentative GA release date.  Without  GA date we can't plan to use AKS with Windows container. We thinking to release our application tentative  March-2020.

**Comment by MikkelHegn** [comment]
The main features we are working on is to integrate Windows node support in the AKS offering, this is yet to be completed and stabilized. We're not able to publicly share a timeline for this work to complete. If you find bugs - please file a support ticket. We do offer support for preview features: https://docs.microsoft.com/en-us/azure/aks/support-policies#preview-features-or-feature-flags

If you find missing features or sub-optimal behavior with the preview, please don't hesitate to open an issue against this list.

**Comment by dariuszbz** [comment]
Would be so useful if you can confirm: end of 2019.

**Comment by sram30** [comment]
Do we have any update on GA date for Windows Container support in AKS?

**Comment by EricMelcher** [comment]
I understand that you can't comment publicly on specifics, but this has been in preview for 6 months, which makes it difficult to plan for. Can someone give any indication of rough timeline? End of 2019? Summer of 2020? Never?

**Comment by dariuszbz** [comment]
Kingdom for the day ....

**Comment by sram30** [comment]
Microsoft Team, I was told that we can expect the Windows Container support in AKS will be available in Q4 2019. We are almost 2/3 into Q4 and still there is no announcement or communication on it. I am surprised that you still haven't figured out the GA date yet. There are a lot of folks awaiting on a response. Appreciate if someone can communicate back to us on this.

**Comment by dariuszbz** [comment]
Kingdom for the date

**Comment by Rodske** [comment]
Any update on this? Any chance that the GA of AKS for Windows will coincide with https://azure.microsoft.com/en-us/updates/azure-container-service-will-retire-on-january-31-2020/ ?

**Comment by adamcarter81** [comment]
Is there any update yet? It would be really helpful for scheduling an application release on AKS with Windows nodes.

**Comment by Rzpeg** [comment]
Is there any ETA for this feature GA? it's 2020 already

**Comment by jjburr** [comment]
any sort of update would be nice.....

**Comment by dariuszbz** [comment]
Can't agree more..... My company is heading to run prod Q1. It would be
very helpful to know the exact date.

**Comment by Rzpeg** [comment]
@jnoller any update?

**Comment by huelsmannt** [comment]
Dear Microsoft-Team,
since here are already many questions when this feature gets GA I won't do the same.

But my question is: are there actually any issues with AKS Windows Node support? Kubernetes has production level support for Windows Nodes since 1.14. Where can I except problems to appear if we use Windows Nodes in productions?

Thanks!

**Comment by dariuszbz** [comment]
I think forget about it and run windows nod/pod in AWS or Google 😃🤪 it
works as GA

**Comment by Rzpeg** [comment]
@huelsmannt Not only Kubernetes has production level support of windows nodes since 1.14, but both competing cloud platforms' managed Kubernetes solutions have them GA. We really need an update on this, @jluk

**Comment by MikkelHegn** [comment]
We appreciate the interest in this feature, but I can only refer back to the comment above: https://github.com/Azure/AKS/issues/980#issuecomment-538943165

Regarding competitive offerings mentioned by @Rzpeg, I assume this refers to EKS and GKE. GKE announced beta support last week:

> In the **beta release** of Windows Server container support in GKE (version 1.16.4)

https://cloud.google.com/blog/products/containers-kubernetes/run-windows-server-containers-on-gke

**Comment by Rzpeg** [comment]
@MikkelHegn
There should have been some sort of update, 5 months have passed, right?
Yes, GKE has it in beta and EKS in GA.
Anyways, thanks for the update.

**Comment by weijuans-msft** [comment]
All - As someone directly driving Windows containers product development and making the technology best for our Azure and on prem customers, I am very excited to see the interest here. While the GA is TBD, if you'd like to talk to us, and share how you are using Windows containers, what are your top pains/asks, we are all ears. Please feel free to contact us at: win-containers@microsoft.com.

**Comment by dariuszbz** [comment]
It was so easy to update this topic... Thank you Aws and Google, without
you they'd say nothing ..... Ahhhh thank you Microsoft for tracking and
giving the feedback.

**Comment by AviateX14** [comment]
Hi, I see the roadmap label has been removed, just over two months since the last update here, are you able to provide any more insight into when this will be GA?

Further, could you provide comment (or a link to existing docs if I've missed them) on:
1. Beyond the standard "it's in preview, be careful", could we get the details of what definitely doesn't work/current limitations?
2. If we were to build our own cluster on IaaS using aks-engine directly (rather than AKS) - would the same limitations/issues apply, or are they only applicable to AKS itself?
3. Is windows node support in aks-engine considered GA/okay to use in production?

Thanks!

**Comment by MikkelHegn** [comment]
@AviateX14 

1. Known limitations in AKS: https://docs.microsoft.com/en-us/azure/aks/windows-node-limitations - note the reference to upstream limitations as well.

2. and 3. AKS-engine support statement: https://github.com/Azure/aks-engine#support - More clarity on difference between AKS and AKS-Engine: https://github.com/Azure/aks-engine/blob/master/docs/faq.md#whats-the-difference-between-aks-and-aks-engine. Upstream limitations are the same between AKS and AKS-Engine, beyond that, as you control all resources, you can probably work around most of the limitation implied by what we support in the managed version of Kubernetes (AKS).

**Comment by brobichaud** [comment]
@MikkelHegn has something delayed Windows support going GA in AKS? I was expecting to see the announcement 2 weeks ago. We really need the full Azure Support backing and SLA with this GA before we can finish migrating our workloads to AKS.

**Comment by dariuszbz** [comment]
Not only you my friend, not only you ......

**Comment by fortanu82** [comment]
Is there any tentative timeline for Windows node to be available with AKS for Production use? Thanks

**Comment by BrainDeveloper** [comment]
I am requesting AKS support windows node with following features:
1. should automatically forwarded the console logs by default so that user can check from azure logs.
2. allow aks owner to log on the windows node machines or allow owner to install some extension on windows nodes.

**Comment by BrainDeveloper** [comment]
As AKS windows node already preview almost one year, what's the plan for shipping to Production ?

**Comment by jluk** [comment]
This has gone GA, you can read about the announcement here:

https://azure.microsoft.com/blog/announcing-the-general-availability-of-windows-server-containers-and-private-clusters-for-azure-kubernetes-service/

Closing this as a result in favor of any requests in new issues.
